### ISSUE TITLE
what do you have against `module.hot`?

### ISSUE THREAD

**Issue by smac89** [import-js/eslint-plugin-import#2328]
Why does the following code trigger this rule?

```ts
if (module.hot) {
    console.warn('module.hot');
}
```
Not doing any `require`, but just by accessing this property, eslint starts complaining. Doesn't make sense, or am I missing something here?

**Comment by ljharb** [comment]
What complaint? Can you provide the warning output?

(Either way, `module.hot` isn't a standard node thing, so relying on it is a super bad idea)

**Comment by smac89** [comment]
There is nothing else other than eslint complaining that this rule has been violated:

The complete error message is this:

> Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')eslint(import/no-import-module-exports)

It prints this error for each line of import inside the file that used the above code

**Comment by smac89** [comment]
Just as I said that, I found the [correct way](https://webpack.js.org/api/hot-module-replacement/) of checking for hot module replacement using es6 syntax:

```
if (import.meta.webpackHot {
    console.warn('module.hot');
}
```

**Comment by ljharb** [comment]
`module` indeed doesn't exist in ESM modules. Glad you figured it out!
